R/utils_render_rtf_redo.R

```diff
@@ -0,0 +1,1240 @@
+      dplyr::select(-matches("^group[s]?$"))
+
+    for (j in seq_len(nrow(summary_df))) {
+
+      cell_list <-
+        lapply(
+          seq_len(n_cols), FUN = function(x) {
+
+            rtf_tbl_cell(
+              rtf_font(
+                font_size = 10,
+                rtf_raw(summary_df[[j, x]])
+              ),
+              h_align = col_alignment[x],
+              borders = list(
+                rtf_border(
+                  "bottom",
+                  color = table_body_hlines_color,
+                  width = ifelse(j == nrow(summary_df), 50, 10)
+                  ),
+                rtf_border("left", color = table_body_vlines_color, width = 10),
+                rtf_border("right", color = table_body_vlines_color, width = 10)
+              )
+            )
+          }
+        )
+
+      row_list_body <-
+        c(
+          row_list_body,
+          rtf_tbl_row(
+            cell_list,
+            height = 0,
+            borders = list(
+              rtf_border(
+                "top",
+                style = ifelse(j == 1, "db", "s"),
+                color = table_body_hlines_color,
+                width = ifelse(j == 1, 50, 10)
+              )
+            ),
+          )
+        )
+    }
+  }
+
+  # Return a list of RTF table rows for the body part
+  row_list_body
+}
+
+#
+# Table Footnotes
+#
+create_footnotes_component_rtf <- function(data) {
+
+  # Get table components and metadata using the `data`
+  footnotes_tbl <- dt_footnotes_get(data)
+
+  # If the `footnotes_tbl` object has no
+  # rows, then return an empty footnotes component
+  if (nrow(footnotes_tbl) == 0) {
+    row_list_footnotes <- list()
+    return(row_list_footnotes)
+  }
+
+  footnotes_tbl <-
+    footnotes_tbl %>%
+    dplyr::select(fs_id, footnotes) %>%
+    dplyr::distinct()
+
+  n_footnotes <- nrow(footnotes_tbl)
+
+  footnote_text <- footnotes_tbl$footnotes
+  footnote_mark <- footnotes_tbl[["fs_id"]]
+
+  footnote_text <-
+    vapply(
+      footnote_text, FUN.VALUE = character(1), USE.NAMES = FALSE,
+      FUN = markdown_to_rtf
+    )
+
+  # # Get the separator option from `opts_df`
+  # separator <-
+  #   opts_df %>%
+  #   dplyr::filter(parameter == "footnotes_sep") %>%
+  #   dplyr::pull(value) %>%
+  #   unlist()
+  #
+  # # Convert common HTML tags/entities to plaintext
+  # separator <-
+  #   separator %>%
+  #   tidy_gsub("<br\\s*?(/|)>", "\n") %>%
+  #   tidy_gsub("&nbsp;", " ")
+
+  # row_list_footnotes <- list()
+  #
+  # for (i in seq_len(n_footnotes)) {
+  #
+  #   cell_list <-
+  #     list(
+  #       rtf_tbl_cell(
+  #         rtf_font(
+  #           italic = TRUE,
+  #           super_sub = "super",
+  #           font_size = 10,
+  #           footnote_mark[i]
+  #         ),
+  #         rtf_font(
+  #           font_size = 10,
+  #           footnote_text[i]
+  #         )
+  #       )
+  #     )
+  #
+  #   row_list_footnotes <- c(row_list_footnotes, rtf_tbl_row(cell_list))
+  # }
```

Comment:
Can this commented out code be deleted now?

Reply:
Just removed it.